phylanx/execution_tree/primitives/variable.hpp

```diff
@@ -24,7 +24,7 @@ namespace phylanx { namespace execution_tree { namespace primitives
     {
     public:
         variable() = default;
-        variable(std::string name);
+        variable(std::string data,std::string name);
```

Comment:
I think , this is not necessary as a `primitive_argument_type` can be constructed from a `std::string` (see https://github.com/STEllAR-GROUP/phylanx/blob/master/phylanx/execution_tree/primitives/base_primitive.hpp#L112-L122).

Reply:
The problem is, if the (std::string name) constructor exists, it somehow gets called. The variable 
 constructed by "return primitive{hpx::local_new<primitives::variable>(d,"")};" gets replaced by it, and the contents of data_ erased.

Reply:
Hmmm, sorry - I don't understand what the problem is. Could you elaborate, please?

Reply:
I can only describe what happens, I can't make sense of it. If a constructor is present that only sets the name, then regardless of how I instantiate the variable I pass to hpx::local_new<primitives::variable>, the name is extracted and a copy of the variable with the data erased is passed into the component. Apparently, nothing currently requires the name-only constructor at present anyway.

Reply:
That sounds like a bug to me. Could you create a small test case that reproduces the issue you're seeing, please?